peachjam_search/models.py

```diff
@@ -77,3 +84,56 @@ class Meta:
     def save(self, *args, **kwargs):
         self.score = exp(-0.1733 * (self.position - 1))
         super().save(*args, **kwargs)
+
+
+class SavedSearch(models.Model):
+    q = models.CharField(max_length=4098)
+    filters = models.CharField(max_length=4098)
+    note = models.TextField(null=True, blank=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    created_at = models.DateField(auto_now_add=True)
+    last_alert = models.DateTimeField(null=True, blank=True)
+
+    def __str__(self):
+        return self.q
+
+    def update_and_alert(self):
+        hits = self.find_new_hits()
+        if hits:
+            self.send_alert(hits)
+            self.last_alert = now()
+            self.save()
+
+    def find_new_hits(self):
+        from peachjam_search.views import DocumentSearchViewSet
+
+        factory = APIRequestFactory()
+        request = factory.get("/search/api/documents/")
+        request.user = self.user
+        params = f"q={self.q}&{self.filters}&created_at__gte={self.last_alert}"
```

Comment:
this won't encode self.q properly. Instead, build up the query dict by setting individual parameters directly so that they get encoded.

Reply:
updated